pkg/dataobj/internal/dataset/page_reader.go

```diff
@@ -296,6 +296,13 @@ func (pr *pageReader) Close() error {
 		pr.closer = nil
 		return err
 	}
+
+	// After closing the reader, we reset the valuesReader to make sure it isn't
+	// holding onto the reference to pr.closer.
+	if pr.valuesReader != nil {
+		pr.valuesReader.Reset(nil)
+	}
```

Comment:
This is just being introduced for safety; without this, a bug which continues to read from the valuesReader after `pr.closer.Close()` may receive bytes from another page. Now, it would return an EOF error.